### PR TITLE
Updating README.md file to be consistent with Python

### PR DESCRIPTION
**Pull request by rangothedog**
Adding information about Entra authentication and a sample using DefaultAzureCredential.

### CODE REVIEW

File: sdk/translation/Azure.AI.Translation.Text/README.md [hunk 1]
```diff
@@ -65,6 +65,26 @@ string region = "<Text Translator Azure Region>";
 TextTranslationClient client = new TextTranslationClient(new AzureKeyCredential(apiKey), new Uri(endpoint), region);
 ```
 
+#### Create `TextTranslationClient` with Entra Credential
```
Comment (jsquire) [inline sdk/translation/Azure.AI.Translation.Text/README.md:68]:
```suggestion
#### Create `TextTranslationClient` with Entra ID
```

**Review by jsquire — APPROVED** [review]

**Comment by jsquire** [comment]
@scottaddie :  Do you mind giving the README a quick pass for the Entra brand usage?

File: sdk/translation/Azure.AI.Translation.Text/README.md (hunk 1 above)
Comment (scottaddie) [inline sdk/translation/Azure.AI.Translation.Text/README.md:68]: Per the product branding guidelines, any occurrence of "Entra" should be prefixed with "Microsoft".

File: sdk/translation/Azure.AI.Translation.Text/README.md [hunk 2]
```diff
@@ -65,6 +65,26 @@ string region = "<Text Translator Azure Region>";
...
+
+```dotnetcli
+dotnet add package Azure.Identity
+```
+
+Create a [custom subdomain][custom_subdomain] for your resource in order to use this type of authentication.  Use this value for the `endpoint` variable for `Text Translator Custom Endpoint`.  Only custom subdomains are supported by the SDK.  Outside of the SDK, you can use Entra authorization for Machine Translation endpoints, as detailed [here][mt_endpoints].
+
+You will also need to [register a new Entra application][register_aad_app] and [grant access][aad_grant_access] to your Translator resource by assigning the `"Cognitive Services User"` role to your service principal.  Additional information about Entra authentication is available [here][custom_details].
+
+Set the values of the `client ID`, `tenant ID`, and `client secret` of the Entra application as environment variables: `AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`.  The DefaultAzureCredential constructor uses these variables to create your credentials.
```
Comment (scottaddie) [inline sdk/translation/Azure.AI.Translation.Text/README.md:80]:
```suggestion
Set the values of the `client ID`, `tenant ID`, and `client secret` of the Entra application as environment variables: `AZURE_CLIENT_ID`, `AZURE_TENANT_ID`, `AZURE_CLIENT_SECRET`.  The `DefaultAzureCredential` constructor uses these variables to create your credentials.
```

**Review by scottaddie — CHANGES_REQUESTED** [review]

File: sdk/translation/Azure.AI.Translation.Text/README.md (hunk 1 above)
Comment (rangothedog) [reply sdk/translation/Azure.AI.Translation.Text/README.md]: Added & pushed.

File: sdk/translation/Azure.AI.Translation.Text/README.md [hunk 3]
```diff
@@ -449,10 +469,15 @@ This project has adopted the [Microsoft Open Source Code of Conduct][code_of_con
 [dictionaryexamples_sample]: https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/translation/Azure.AI.Translation.Text/samples/Sample6_DictionaryExamples.md
 
 [translator_resource_create]: https://learn.microsoft.com/azure/cognitive-services/Translator/create-translator-resource
-
+[azure_identity]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/identity/Azure.Identity/README.md
+[DefaultAzureCredential]: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/README.md#defaultazurecredential
+[register_aad_app]: https://learn.microsoft.com/azure/cognitive-services/authentication#assign-a-role-to-a-service-principal
+[aad_grant_access]: https://learn.microsoft.com/azure/cognitive-services/authentication#assign-a-role-to-a-service-principal
```
[the remark this answers is not in the document]
Comment (rangothedog) [reply sdk/translation/Azure.AI.Translation.Text/README.md:475]: Fixed & pushed!

**Review by scottaddie — APPROVED** [review]

**Review by MikeyMCZ — APPROVED** [review]

**Pull request merged**